shadowsocks/salt_generator.go

```diff
@@ -64,23 +78,28 @@ var serverSaltLabel = []byte("outline-server-salt")
 // random, but is secretly marked as being issued by the server.
 // This is useful to prevent the server from accepting its own output in a
 // reflection attack.
-func NewServerSaltGenerator(secret string) *ServerSaltGenerator {
+func NewServerSaltGenerator(cipher shadowaead.Cipher, secret string) ServerSaltGenerator {
+	if cipher.SaltSize()-markLen < minEntropy {
```

Comment:
Can you move the implementation choice to the MakeCipherEntry instead?

It will improve the readability of MakeCipherEntry, since it will make it clearer that we use different implementations depending on the cipher. Otherwise it looks like it's always the same behavior.

It will also keep this code lower level and make it more reusable, since the caller can now choose whatever logic they want. `minEntropy` can move to the caller, since that's application logic.

Reply:
Done